pkg/assembler/backends/ent/backend/pkgequal_test.go

```diff
@@ -544,6 +544,209 @@ func (s *Suite) TestPkgEqual() {
 	}
 }
 
+func (s *Suite) TestIngestPkgEquals() {
+	type call struct {
+		P1 []*model.PkgInputSpec
+		P2 []*model.PkgInputSpec
+		PE []*model.PkgEqualInputSpec
+	}
+	tests := []struct {
+		Name         string
+		InPkg        []*model.PkgInputSpec
+		Calls        []call
+		Query        *model.PkgEqualSpec
+		ExpHE        []*model.PkgEqual
+		ExpIngestErr bool
+		ExpQueryErr  bool
+	}{
+		{
+			Name:  "HappyPath",
+			InPkg: []*model.PkgInputSpec{p1, p2},
+			Calls: []call{
+				{
+					P1: []*model.PkgInputSpec{p1, p1},
+					P2: []*model.PkgInputSpec{p2, p2},
+					PE: []*model.PkgEqualInputSpec{
+						{
+							Justification: "test justification",
+						},
+						{
+							Justification: "test justification",
+						},
+					},
+				},
+			},
+			Query: &model.PkgEqualSpec{
+				Justification: ptrfrom.String("test justification"),
+			},
+			ExpHE: []*model.PkgEqual{
+				{
+					Packages:      []*model.Package{p1out, p2out},
+					Justification: "test justification",
+				},
+			},
+		},
+		{
+			Name:  "Ingest same, different order",
+			InPkg: []*model.PkgInputSpec{p1, p2},
+			Calls: []call{
+				{
+					P1: []*model.PkgInputSpec{p1, p2},
+					P2: []*model.PkgInputSpec{p2, p1},
+					PE: []*model.PkgEqualInputSpec{
+						{
+							Justification: "test justification",
+						},
+						{
+							Justification: "test justification",
+						},
+					},
+				},
+			},
+			Query: &model.PkgEqualSpec{
+				Justification: ptrfrom.String("test justification"),
+			},
+			ExpHE: []*model.PkgEqual{
+				{
+					Packages:      []*model.Package{p1out, p2out},
+					Justification: "test justification",
+				},
+			},
+		},
+		{
+			Name:  "Query on pkg details",
+			InPkg: []*model.PkgInputSpec{p1, p2, p3},
+			Calls: []call{
+				{
+					P1: []*model.PkgInputSpec{p1, p1},
+					P2: []*model.PkgInputSpec{p2, p3},
+					PE: []*model.PkgEqualInputSpec{
+						{
+							Justification: "test justification",
+						},
+						{
+							Justification: "test justification",
+						},
+					},
+				},
+			},
+			Query: &model.PkgEqualSpec{
+				Packages: []*model.PkgSpec{{
+					Version: ptrfrom.String("2.11.1"),
+					Subpath: ptrfrom.String(""),
+				}},
+			},
+			ExpHE: []*model.PkgEqual{
+				{
+					Packages:      []*model.Package{p1out, p2out},
+					Justification: "test justification",
+				},
+			},
+		},
+		{
+			Name:  "Query on pkg algo and pkg",
+			InPkg: []*model.PkgInputSpec{p1, p2, p3},
+			Calls: []call{
+				{
+					P1: []*model.PkgInputSpec{p1, p1},
+					P2: []*model.PkgInputSpec{p2, p3},
+					PE: []*model.PkgEqualInputSpec{
+						{
+							Justification: "test justification",
+						},
+						{
+							Justification: "test justification",
+						},
+					},
+				},
+			},
+			Query: &model.PkgEqualSpec{
```

Comment:
hmm is this a bug on the `inmem` backend that it does not return all if the `subpath` is not set?

Reply:
I think so based on consideration that lacking a field in the query's spec should mean "don't filter by this field"

Reply:
Draft PR to reproduce the issue https://github.com/guacsec/guac/pull/1249

Reply:
yeah let's discuss this there. No need to block this PR.